### PR TITLE
TL/CUDA: multiring allgatherv and reduce_scatterv

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
## What
Support multiple rings in TL CUDA allgather and allgatherv

### CODE REVIEW

**Comment by Sergei-Lebedev** [comment]
added multiring reduce_scatterv

File: src/components/tl/cuda/tl_cuda_lib.c [hunk 1]
```diff
@@ -15,6 +15,11 @@ UCC_CLASS_INIT_FUNC(ucc_tl_cuda_lib_t, const ucc_base_lib_params_t *params,
     UCC_CLASS_CALL_SUPER_INIT(ucc_tl_lib_t, &ucc_tl_cuda.super,
                               &tl_config->super);
     memcpy(&self->cfg, tl_config, sizeof(*tl_config));
+    if (self->cfg.allgather_ring_max_rings > UCC_TL_CUDA_MAX_RINGS) {
```
Comment (vspetrov) [inline src/components/tl/cuda/tl_cuda_lib.c:18]: same check for reduce_scatter_max_rings ?

**Review by vspetrov — APPROVED** [review]
@Sergei-Lebedev I tried to look carefully and it all looks good to me (just minor comment below). 👍

**Review by manjugv — APPROVED** [review]

File: src/components/tl/cuda/tl_cuda.h [hunk 2]
```diff
@@ -22,6 +22,8 @@
 #endif
 
 #define UCC_TL_CUDA_MAX_PEERS 8
+#define UCC_TL_CUDA_MAX_RINGS 8
```
Comment (bureddy) [inline src/components/tl/cuda/tl_cuda.h:25]: will be any scenario in the future where we want to try mode than 8 rings? instead should we allocate memory dynamically

**Pull request merged**